File: c++/seleccion/SeleccionTest.cpp
#include <iostream>

using namespace std;
int main(int argc, char *argv[]) {
	int datos[] = {8,5,3,9,1};
		
	cout << "Arreglo original: [";
	copy(begin(datos), end(datos), ostream_iterator<int>(cout, ","));
	cout << "\b]" << endl;
	
	int length = sizeof(datos) / sizeof(datos[0]); //Calcula el # de elementos
	
	for (int indiceOrdenar = 0; indiceOrdenar < length - 1; indiceOrdenar++) {
		int indiceMinimo = indiceOrdenar;
		for (int indiceComparar = indiceOrdenar + 1; indiceComparar < length; indiceComparar++) {
			if ( datos[indiceComparar] < datos[indiceMinimo] ) {
				indiceMinimo = indiceComparar;
			}
		}
		int buffer = datos[indiceOrdenar];
		datos[indiceOrdenar] = datos[indiceMinimo];
		datos[indiceMinimo] = buffer;
	}
			
	cout << "Arreglo ordenado: [";
	copy(begin(datos), end(datos), ostream_iterator<int>(cout, ","));
	cout << "\b]";
	
	return 0;
}

File: python/2.2 Algoritmos de Ordenamiento/seleccion/SeleccionTest.py
import numpy as np

datos = np.array([8,5,3,9,1])
print ("Arreglo original: " + str(datos))

for indiceOrdenar in range(0, datos.size -1, 1):
	indiceMinimo = indiceOrdenar 
	for indiceComparar in range(indiceOrdenar + 1, datos.size, 1): 				
		if (datos[indiceComparar] < datos[indiceMinimo]): 
			indiceMinimo = indiceComparar
		
		buffer = datos[indiceOrdenar] # Intercambio
		datos[indiceOrdenar] = datos[indiceMinimo]
		datos[indiceMinimo] = buffer

print ("Arreglo ordenado: " + str(datos))

